Week8/woojo230/mission01/src/pages/search.tsx

```diff
@@ -0,0 +1,129 @@
+    hasNextPage,
+    isFetching,
+    isLoading,
+    error,
+    refetch,
+  } = useGetInfiniteScroll({
+    order: order,
+    limit: 50,
+    search: searchType === 'title' ? debouncedSearch : '',
+  });
+
+  const {
+    data: tagData,
+    fetchNextPage: tagFetchNextPage,
+    hasNextPage: tagHasNextPage,
+    isFetching: tagIsFetching,
+    isLoading: tagIsLoading,
+    error: tagError,
+    refetch: tagRefetch,
+  } = useGetTagInfiniteScroll({
+    tagName: searchType === 'tag' ? debouncedSearch : '',
+    order: order,
+    limit: 50,
+    search: '',
+    enabled: searchType === 'tag' && debouncedSearch !== '',
+  });
+
+  const allLpData = lpData?.pages.flatMap((page) => page.data) ?? [];
+  const allTagData = tagData?.pages.flatMap((page) => page.data) ?? [];
+
+  const observerRef = useInfiniteScrollObserver(
+    searchType === 'title' ? hasNextPage : tagHasNextPage,
+    searchType === 'title' ? isFetching : tagIsFetching,
+    searchType === 'title' ? fetchNextPage : tagFetchNextPage
+  );
+
+  const handleClick = () => {
+    if (searchType === 'title') {
+      setSearchType('tag');
+      setSearch('');
+    } else {
+      setSearchType('title');
+      setSearch('');
+    }
+  };
+
+  useEffect(() => {
+    const currentInput = inputRef.current;
+    if (currentInput) {
+      const { selectionStart, selectionEnd } = currentInput;
+      if (searchType === 'title') {
+        refetch().then(() => {
+          currentInput.focus();
+          currentInput.setSelectionRange(selectionStart, selectionEnd);
+        });
+      } else {
+        tagRefetch().then(() => {
+          currentInput.focus();
+          currentInput.setSelectionRange(selectionStart, selectionEnd);
+        });
+      }
+    } else {
+      if (searchType === 'title') {
+        refetch();
+      } else {
+        tagRefetch();
+      }
+    }
+  }, [debouncedSearch, refetch, tagRefetch, searchType]);
+
+  if (searchType === 'title' ? isLoading : tagIsLoading)
+    return <LoadingSpinner />;
+  if (searchType === 'title' ? error : tagError) return <Error />;
+
+  return (
+    <div className="flex flex-col items-center min-h-screen">
+      <div className="w-1/2 flex flex-col gap-4 py-4 sticky top-0 bg-white z-10">
+        <div className="flex gap-2">
+          <SearchForm
+            inputRef={inputRef}
+            searchType={searchType}
+            search={search}
+            setSearch={setSearch}
+          />
+        </div>
+        <div className="flex gap-2">
+          <TagBtn searchType={searchType} handleClick={handleClick} />
+        </div>
+      </div>
+      <OrderButtons
+        order={order}
+        onAscClick={() => setOrder('asc')}
+        onDescClick={() => setOrder('desc')}
+      />
+      <LpGrid
+        lpData={searchType === 'title' ? allLpData : allTagData}
+        isFetching={searchType === 'title' ? isFetching : tagIsFetching}
+        isLoading={searchType === 'title' ? isLoading : tagIsLoading}
+      />
```

Comment:
아래 예시처럼 데이터 표시 부분만 조건부 렌더링으로 변경하면, 로딩 중에도 검색창과 버튼은 유지되어 
`input` 포커스가 사라지는 문제도 해결됩니다.


```suggestion
    {(searchType === "title" ? isLoading : tagIsLoading) ? (
      <LoadingSpinner />
    ) : (
      <LpGrid
        lpData={searchType === 'title' ? allLpData : allTagData}
        isFetching={searchType === 'title' ? isFetching : tagIsFetching}
        isLoading={searchType === 'title' ? isLoading : tagIsLoading}
      />
    )}
```

Reply:
수정완